migration/rails.rst

```diff
@@ -0,0 +1,222 @@
+into your Gemfile, running ``bundle install``, and then annotating your
+ActiveRecord models like this:
+
+.. code:: ruby
+
+  class PageView < ActiveRecord::Base
+    multi_tenant :customer
+    # ...
+  end
+
+In this case ``customer`` is the tenant model, and your ``page_views``
+table needs to have a ``customer_id`` column that references the
+customer the page view belongs to.
+
+The `activerecord-multi-tenant
+<https://github.com/citusdata/activerecord-multi-tenant>`__ Gem aims to
+make it easier to implement the above data changes in a typical Rails
+application.
+
+As mentioned in the beginning, by adding ``multi_tenant :customer``
+annotations to your models, the library automatically takes care of
+including the tenant\_id with all queries.
+
+In order for that to work, you’ll always need to specify which tenant
+you are accessing, either by specifying it on a per-request basis:
+
+.. code:: ruby
+
+  class ApplicationController < ActionController::Base
+    # Opt-into the "set_current_tenant" controller helpers by specifying this:
+    set_current_tenant_through_filter
+
+    before_filter :set_customer_as_tenant
+
+    def set_customer_as_tenant
+      customer = Customer.find(session[:current_customer_id])
+      set_current_tenant(customer) # Set the tenant
+    end
+  end
+
+Or by wrapping your code in a block, e.g. for background and maintenance
+tasks:
+
+.. code:: ruby
+
+  customer = Customer.find(session[:current_customer_id])
+  # ...
+  MultiTenant.with(customer) do
+    site = Site.find(params[:site_id])
+
+    # Modifications automatically include tenant_id
+    site.update! last_accessed_at: Time.now
+
+    # Queries also include tenant_id automatically
+    site.page_views.count
+  end
+
+Once you are ready to use a distributed multi-tenant database like
+Citus, all you need is a few adjustments to your migrations, and you're
+good to go:
+
+.. code:: ruby
+
+  class InitialTables < ActiveRecord::Migration
+    def up
+      create_table :page_views, partition_key: :customer_id do |t|
+        t.references :customer, null: false
+        t.references :site, null: false
+
+        t.text :url, null: false
+        ...
+        t.timestamps null: false
+      end
+      create_distributed_table :page_views, :account_id
+    end
+
+    def down
+      drop_table :page_views
+    end
+  end
+
+Note the ``partition_key: :customer_id``, something that's
+added to Rails' ``create_table`` by our library, which ensures
+that the primary key includes the tenant\_id column, as well as
+``create_distributed_table`` which enables Citus to scale out the data
+to multiple nodes.
+
+Example Application
+-------------------
+
+If you are interested in a more complete
+example, check out our `reference app
+<https://github.com/citusdata/citus-example-ad-analytics>`__ that
+showcases a simplified sample SaaS application for ad analytics.
+
+.. figure:: ../images/ad_analytics_screenshot.png
```

Comment:
Ha, discovered what it was - the filename triggered ad blockers! I was confused for a while, thought it was something about the build pipeline.